### PR TITLE
Paypal payments endpoints

### PR DESCRIPTION
**Pull request by eshaibu**

### CODE REVIEW

File: src/payment-execute.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  try {
+    await configurePayPal(data);
+
+    checkRequestMethodType(requestMethod, ['POST'], actionsMessage);
+
+    validateRequired({ payer_id, payment_id });
+
+    const { statusCode, paypalResponse } = await callEndpoint('execute', { payer_id, transactions }, payment_id);
+    response.json(paypalResponse, statusCode);
```
Comment (adamwardecki) [inline src/payment-execute.js]: `return response.json(...`

(fix across the PR)

File: src/payments.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  try {
+    await configurePayPal(data);
+
+    checkRequestMethodType(requestMethod, expectedMethodTypes, actionsMessage);
+
+    if (requestMethod === 'POST') {
+      validateRequired({ create_payment_details });
+      const { statusCode, paypalResponse } = await callEndpoint('create', create_payment_details);
+      response.json(paypalResponse, statusCode);
```
Comment (adamwardecki) [inline src/payments.js]: return response

File: test/payments.test.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  });
+
+  describe('Invalid request method', () => {
+    it('should return an error if request method is of type `DELETE`', (done) => {
+      meta.request.REQUEST_METHOD = 'DELETE';
+      run('payments', { meta })
+        .then((res) => {
+          const actionsMessage = 'creating, retrieving and updating payments respectively';
+          const expectedMethodTypes = ['POST', 'GET', 'PATCH'].join(', ');
+          const errorMessage = `Make sure to use ${expectedMethodTypes} for ${actionsMessage}.`;
```
Comment (adamwardecki) [inline test/payments.test.js]: are these 3 constants necessary? You could put a string directly into the assertion.

File: test/payments.test.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            .to.equal(updated_insurance);
+          done();
+        })
+        .catch((err) => {
+          done(err);
+        });
+    });
+  });
+
+  describe('GET', () => {
```
Comment (adamwardecki) [inline test/payments.test.js:91]: Add test for invalid payment id

File: test/payments.test.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+          done();
+        })
+        .catch((err) => {
+          done(err);
+        });
+    });
+
+    it('should return "VALIDATION_ERROR" if payer parameter absent', (done) => {
+      meta.request.REQUEST_METHOD = 'POST';
+      const argsValidation = { ...create_payment_params, payer: null };
```
Comment (adamwardecki) [inline test/payments.test.js]: why not naming the const `create_payment_details`?

File: test/payments.test.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+import { create_payment_params, update_payment_params, updated_insurance } from './utils/helpers';
+
+describe('payments', () => {
+  const { PAYPAL_CONFIG_URL, PAYPAL_MODE, PAYPAL_CLIENT_ID, PAYPAL_CLIENT_SECRET } = process.env;
+  const meta = generateMeta('payments');
+  let payment_id = '';
+  const requestUrl = request(PAYPAL_CONFIG_URL);
+
+  before((done) => {
+    requestUrl.post('/')
```
Comment (adamwardecki) [inline test/payments.test.js]: why not `request(PAYPAL_CONFIG_URL).post('/')` ?

File: src/payments.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        result = await callEndpoint('list', list_details);
+      }
+      const { statusCode, paypalResponse } = result;
+      response.json(paypalResponse, statusCode);
+    } else if (requestMethod === 'PATCH') {
+      validateRequired({ update_payment_details });
+      const { statusCode, paypalResponse } = await callEndpoint('update', update_payment_details, payment_id);
+      response.json(paypalResponse, statusCode);
+    }
+  } catch ({ error, message, statusCode, details }) {
```
Comment (adamwardecki) [inline src/payments.js]:
```js
catch({ error, ...errorDetails, statusCode = 400 }) {
    return response.json(error || errorDetails, statusCode)
}
```

File: src/payments.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  callEndpoint, checkRequestMethodType, configurePayPal,
+  validateRequired
+} from './utils/helpers';
+
+export default async (ctx) => {
+  const { data, response } = new Syncano(ctx);
+  const requestMethod = ctx.meta.request.REQUEST_METHOD;
+  const { create_payment_details, update_payment_details, payment_id, ...list_details } = ctx.args;
+  const actionsMessage = 'creating, retrieving and updating payments respectively';
+  const expectedMethodTypes = ['POST', 'GET', 'PATCH'];
```
Comment (adamwardecki) [inline src/payments.js]: no need for this const

**Review by adamwardecki — CHANGES_REQUESTED** [review]
Looks good!
Please see the comments

File: test/payments.test.js (hunk 4 above)
Comment (eshaibu) [reply test/payments.test.js:91]: done

**Review by adamwardecki — APPROVED** [review]

**Pull request merged**